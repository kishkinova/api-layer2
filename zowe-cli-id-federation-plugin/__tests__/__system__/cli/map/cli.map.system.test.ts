/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import * as path from "path";
import * as fs from "fs";
import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../__src__/environment/doc/ITestPropertiesSchema";

// Test environment will be populated in the "beforeAll"
let TEST_ENVIRONMENT: ITestEnvironment<ITestPropertiesSchema>;
const testCsv = "users.csv";

describe("id-federation map", () => {
    let csv: string;
    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            installPlugin: true,
            testName: "map_command"
        });
        csv = path.join(TEST_ENVIRONMENT.workingDir, testCsv);
        fs.copyFileSync(path.join(__dirname, "__resources__", testCsv), csv);
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should print the successful creation message", () => {
        const response = runCliScript(__dirname + "/__scripts__/map.sh", TEST_ENVIRONMENT,
            [`${TEST_ENVIRONMENT.workingDir}/users.csv`, "ACF2", "TST1", "ldap://12.34.56.78:910"]);

        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });

    it("should display the help", () => {
        const response = runCliScript(__dirname + "/__scripts__/map_help.sh", TEST_ENVIRONMENT);

        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });
});
