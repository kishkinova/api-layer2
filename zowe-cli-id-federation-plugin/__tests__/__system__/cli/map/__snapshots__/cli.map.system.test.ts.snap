// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`id-federation map should display the help 1`] = `
"
 COMMAND NAME
 ------------

   map

 DESCRIPTION
 -----------

   Generate a JCL based on [inputFile] that creates a mapping association between a
   mainframe user ID and a distributed user identity

 USAGE
 -----

   zowe id-federation map <input-file> [options]

 POSITIONAL ARGUMENTS
 --------------------

   input-file		 (string)

      File with the list of users

 IDF CONNECTION OPTIONS
 ----------------------

   --esm | -e (string)

      The ESM to execute command

      Allowed values: RACF, TSS, ACF2

   --lpar | -l (string)

      The security domain on which command will be executed

   --registry | -r (string)

      The registry that contains the distributed-identity user name

 PROFILE OPTIONS
 ---------------

   --id-federation-profile | --id-federation-p (string)

      The name of a (id-federation) profile to load for this command execution.

   --base-profile | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --host | -H (string)

      Host name of service on the mainframe.

   --port | -P (number)

      Port number of service on the mainframe.

   --user | -u (string)

      User name to authenticate to service on the mainframe.

   --password | --pass | --pw (string)

      Password to authenticate to service on the mainframe.

   --reject-unauthorized | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --token-type | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value | --tv (string)

      The value of the token to pass to the API.

   --cert-file (local file path)

      The file path to a certificate file to use for authentication

   --cert-key-file (local file path)

      The file path to a certificate key file to use for authentication

 GLOBAL OPTIONS
 --------------

   --show-inputs-only (boolean)

      Show command inputs and do not run the command

   --response-format-json | --rfj (boolean)

      Produce JSON formatted data from a command

   --help | -h (boolean)

      Display help text

   --help-web | --hw (boolean)

      Display HTML help in browser

"
`;

exports[`id-federation map should print input arguments 1`] = `
"Input file: file.csv
ESM: TSS
LPAR: TST1
Registry: ldap://12.34.56.78:910
"
`;
